hicommonwealth/commonwealth
# Fix manage page admins list

## Link to Issue
Closes: #4045
Closes: #3985

## Description of Changes
- Adds a new `/roles` endpoint to correctly list roles by chain ID and permissions filter
- Remove deprecated getRoles endpoint
- Updated UI to use updated (non-external) URL

## Test Plan
- Navigate to the Manage page of a community you are an admin/mod for– confirm that the admins list renders as expected
- Click the `X` next to a mod/admin and attempt to remove them– confirm that the user is converted to a member

## Deployment Plan
N/A

## Other Considerations
- With there being some deprecated tables/endpoints, it would be great to have a guide on how to correctly set myself as admin to a community in the local environment. I *thought* I did it properly before, but apparently not.
- I can confirm that a request to `http://localhost:8080/api/roles?chain_id=timelessfinance&permissions[]=moderator&permissions[]=admin` returns the admins for that chain, so it *should* render in the UI properly.
